cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java

```diff
@@ -416,6 +424,35 @@ public void createSchema(PhysicalPlan plan) throws MetadataException {
     }
   }
 
+  private void verifyCreatedSgSuccess(List<PartialPath> deviceIds) throws MetadataException {
+    long startTime = System.currentTimeMillis();
+    while (true) {
+      int successNum = deviceIds.size();
+      for (PartialPath deviceId : deviceIds) {
+        PartialPath storageGroupName = MetaUtils
+          .getStorageGroupPathByLevel(deviceId, IoTDBDescriptor.getInstance()
+            .getConfig().getDefaultStorageGroupLevel());
+        if (IoTDB.metaManager.isStorageGroup(storageGroupName)) {
+          successNum --;
+        }
+      }
```

Comment:
All deviceIds are verified in every loop, which is unnecessary.
It would be better to copy deviceIds into another list and remove them once verified, or use a boolean array to indicate whether the deviceId has been verified.